### PR TITLE
os/binfmt: Add missing macros for elf_dumpbuffer

### PR DESCRIPTION
**Pull request by sangam-swami**
Add missing debug macros for CONFIG_ELF_DUMPBUFFER

Signed-off-by: Sangamanatha <sangam.swami@samsung.com>

### CODE REVIEW

File: os/binfmt/elf.c [hunk 1]
```diff
@@ -74,11 +74,11 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* CONFIG_DEBUG, CONFIG_DEBUG_INFO, and CONFIG_DEBUG_BINFMT have to be
+/* CONFIG_DEBUG, CONFIG_DEBUG_VERBOSE, and CONFIG_DEBUG_BINFMT have to be
  * defined or CONFIG_ELF_DUMPBUFFER does nothing.
  */
 
-#if !defined(CONFIG_DEBUG_INFO) || !defined(CONFIG_DEBUG_BINFMT)
+#if !defined(CONFIG_DEBUG_VERBOSE) || !defined(CONFIG_DEBUG_BINFMT)
```
Comment (Taejun-Kwon) [inline os/binfmt/elf.c:81]: Use CONFIG_DEBUG_BINFMT_INFO stand alone is not enough?

Comment (sangam-swami) [reply os/binfmt/elf.c]: done

File: os/binfmt/elf.c [hunk 2]
```diff
@@ -74,11 +74,11 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-/* CONFIG_DEBUG, CONFIG_DEBUG_INFO, and CONFIG_DEBUG_BINFMT have to be
+/* CONFIG_DEBUG, CONFIG_DEBUG_VERBOSE, and CONFIG_DEBUG_BINFMT have to be
  * defined or CONFIG_ELF_DUMPBUFFER does nothing.
  */
 
-#if !defined(CONFIG_DEBUG_INFO) || !defined(CONFIG_DEBUG_BINFMT)
+#if !defined(CONFIG_DEBUG_BINFMT_INFO)
```
Comment (sunghan-chang) [inline os/binfmt/elf.c:81]: Is this necessary to be two steps?
```c
#if !defined(CONFIG_DEBUG_BINFMT_INFO)
#undef CONFIG_ELF_DUMPBUFFER
#endif

#ifdef CONFIG_ELF_DUMPBUFFER
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
```

How about checking it directly?
```c
#ifdef CONFIG_DEBUG_BINFMT_INFO
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
```

Comment (sangam-swami) [reply os/binfmt/elf.c:81]: if CONFIG_DEBUG_BINFMT_INFO enabled and CONFIG_ELF_DUMPBUFFER is not, then it should not set elf_dumpbuffer, so I think the current approach is fine

Comment (sangam-swami) [reply os/binfmt/elf.c:81]: or we can replace with :
#ifdef CONFIG_DEBUG_BINFMT_INFO
#ifdef CONFIG_ELF_DUMPBUFFER
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
#endif

Comment (sunghan-chang) [reply os/binfmt/elf.c:81]: os/binfmt/libelf/Kconfig
```kconfig
config ELF_DUMPBUFFER
	bool "Dump ELF buffers"
	default n
	depends on DEBUG_BINFMT_INFO  <---- here
	---help---
		Dump various ELF buffers for debug purposes
```

in here, os/binfmt/elf.c
```c
remove 1st conditional  <--- here

#ifdef CONFIG_ELF_DUMPBUFFER
#define elf_dumpbuffer(m, b, n) binfodumpbuffer(m, b, n)
#else
#define elf_dumpbuffer(m, b, n)
#endif
```

This is enough and is the same as current pr

Comment (sangam-swami) [reply os/binfmt/elf.c]: done

**Pull request merged**